spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -240,8 +245,38 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
     }
   }
 
+  public boolean isHiveCatalogConfigValid(CaseInsensitiveStringMap options, StringBuilder errorMsg) {
+    String hadoopConfUri = SparkSession.active().sparkContext().conf().contains("spark.hadoop.hive.metastore.uris") ?
+            SparkSession.active().sparkContext().conf().get("spark.hadoop.hive.metastore.uris") : null;
+    String catalogConfUri = options.get("uri");
+
+    if (hadoopConfUri == null) {
+      errorMsg.append("Hive uri config is missing");
+    } else if (catalogConfUri != null) {
+      LOG.warn("Don't set uri for SparkSessionCatalog" +
+              "set it only in hive conf");
+      if (!hadoopConfUri.equalsIgnoreCase(catalogConfUri)) {
+        errorMsg.append("Cannot set uri for SparkSessionCatalog: " +
+                "conflicts with Hive conf URI");
```

Comment:
Nit: I think it would be more beneficial to the user if the error message contains the two conflicting values, indicating what value was set for the catalog's `uri` field and what value was set via the traditional mechanisms available to spark.